File: assets/codes_sample/react/stale.go
package main

import "fmt"

var	i int =0

func gen() func() {
	i+=1
	n := i
	return func () {
		fmt.Println(n)
	}
}

func main() {
	fn1 := gen()
	fn2 := gen()
	fn1()
	fn2()
}

File: assets/codes_sample/react/stale.js
function classic() {
  const fn = []
  // for (let i = 0; i < 10; i++) {
  for (var i = 0; i < 10; i++) {
    fn.push(function () {
      console.log(i)
    })
  }
  return fn
}
// classic().forEach(item => item())

let i = 0
function genFunc() {
  i++
  let n = i
  return function () {
    console.log(n)
  }
}

const fn1 = genFunc();
const fn2 = genFunc();
fn1()
fn2()